expirationwatch/expiration_watcher.go

```diff
@@ -84,49 +72,9 @@ func (w *Watcher) Remove(expirationTimestamp time.Time, id string) {
 	}
 }
 
-// Watch starts the expiration watchers poller. It continuously checks all items
-// for expiration until there is an error or the given context is canceled. You
-// usually want to call Watch inside a goroutine.
-func (w *Watcher) Watch(ctx context.Context, pollingInterval time.Duration) error {
-	w.mu.Lock()
-	if w.wasStartedOnce {
-		w.mu.Unlock()
-		return errors.New("Can only start Watcher once per instance")
-	}
-	w.wasStartedOnce = true
-	w.mu.Unlock()
-
-	// TODO(fabio): Optimize this poller. We could keep track of soonestExpirationTime as a property of
-	// Watcher. Whenever a new item is added via Add, we check if the expiration time is sooner
-	// than soonestExpirationTime and if so, we update soonestExpirationTime. Then instead of running the
-	// inner for loop at a constant frequency, we adjust the frequency based on the value of
-	// soonestExpirationTime (probably by using time.After or time.Sleep).
-	ticker := time.NewTicker(pollingInterval)
-	for {
-		select {
-		case <-ctx.Done():
-			ticker.Stop()
-			close(w.expiredItems)
-			return nil
-		case <-ticker.C:
-			expiredItems := w.prune()
-			if len(expiredItems) > 0 {
-				w.expiredItems <- expiredItems
-			}
-		}
-	}
-}
-
-// ExpiredItems returns a read-only channel that can be used to listen for
-// expired items. The channel will be closed if/when the watcher is done
-// watching.
-func (w *Watcher) ExpiredItems() <-chan []ExpiredItem {
-	return w.expiredItems
-}
-
-// prune checks for any expired items, removes them from the expiration watcher and returns them
-// to the caller
-func (w *Watcher) prune() []ExpiredItem {
+// Prune checks for any expired items given the latest blockTimestamp,
```

Comment:
`expirationwatch.Watcher` is used for other things besides block timestamps right? Can we rename this parameter to reflect that?